lib/LedgerSMB/PriceMatrix.pm

```diff
@@ -173,6 +124,11 @@ sub price_matrix {
                            - ($sellprice * ($mref->{pricebreak} / 100));
             }
             $ref->{sellprice} = $sellprice;
+            if ($mref->{qty} > $form->{qtycache}->{$ref->{id}}){
```

Comment:
@einhverfr just ran into a problem with the quantity cache, I think: when loading an existing invoice from the database, the price matrix is applied to each item individually instead of running the pricematrix at the end.

(which explains part of the cases of 'rowcount' being '' or undefined...